Firebase/Core/Public/FIRTimestamp.h

```diff
@@ -19,41 +19,29 @@
 NS_ASSUME_NONNULL_BEGIN
 
 /**
- * An FSTTimestamp represents an absolute time from the backend at up to nanosecond precision.
- * An FSTTimestamp is represented in terms of UTC and does not have an associated timezone.
+ * An FIRTimestamp represents an absolute time from the backend at up to microsecond precision.
+ * An FIRTimestamp is represented in terms of UTC and does not have an associated timezone.
  */
-@interface FSTTimestamp : NSObject <NSCopying>
+NS_SWIFT_NAME(Timestamp)
+@interface FIRTimestamp : NSObject <NSCopying>
 
+/** */
 - (instancetype)init NS_UNAVAILABLE;
 
 /**
  * Creates a new timestamp.
  *
  * @param seconds the number of seconds since epoch.
- * @param nanos the number of nanoseconds after the seconds.
+ * @param nanoseconds the number of nanoseconds after the seconds.
  */
-- (instancetype)initWithSeconds:(int64_t)seconds nanos:(int32_t)nanos NS_DESIGNATED_INITIALIZER;
-
-/** Creates a new timestamp with the current date / time. */
-+ (instancetype)timestamp;
++ (instancetype)timestampWithSeconds:(int64_t)seconds nanoseconds:(int32_t)nanoseconds;
 
 /** Creates a new timestamp from the given date. */
 + (instancetype)timestampWithDate:(NSDate *)date;
```

Comment:
@ryanwilson Are we really going to insist on these factory functions rather than just exposing initializers? For swift users this makes this type feel foreign.

I vote we get rid of the factory functions and just expose the initializers.

Reply:
I have no objection to making these initializers - although Swift should automatically make these factory methods initializers unless we explicitly add markup to prevent that from happening. If it follows the pattern: `FIRFoo` and `+ (instancetype)fooWithBar:...` it translates to `init(bar:...)`.

Xcode generated this for me:
<img width="911" alt="Automatic factory to initializer conversion" src="https://user-images.githubusercontent.com/1097316/36218717-4fe45264-1183-11e8-8cd4-9767d1145afb.png">

Reply:
Ah, in that case I take this feedback back.

Reply:
@wilhuff, can/should I add some tests for Timestamp to Example/SwiftBuildTest/main.swift?

Reply:
It wouldn't hurt, but probably isn't worth it. We're about to add a bunch of actual tests for these public types via changes like #815 and there it's an actual test of behavior in Swift which is even more than our "does it build" demonstration in `main.swift`.